wfe/wfe.go

```diff
@@ -1356,22 +1345,37 @@ func (wfe *WebFrontEndImpl) updateChallenge(
 		return
 	}
 
-	var chalResp acme.Challenge
+	var chalResp struct {
+		KeyAuthorization string
+	}
 	err := json.Unmarshal(body, &chalResp)
 	if err != nil {
 		wfe.sendError(
 			acme.MalformedProblem("Error unmarshaling body JSON"), response)
 		return
 	}
 
+	// Historically challenges were updated by POSTing a KeyAuthorization. This is
+	// unnecessary, the server can calculate this itself. We could ignore this if
+	// sent (and that's what Boulder will do) but for Pebble we'd like to be more
+	// aggressive about pushing clients implementations in the right direction, so
+	// we treat this as a malformed request.
+	if chalResp.KeyAuthorization != "" {
```

Comment:
Let's make this field a pointer so we can check if it's present at all. Otherwise we'll accept its presence with an empty string.

Reply:
Good idea, thanks!

Reply:
Resolved in ee20d06